### ISSUE TITLE
The loss_gradient of PyTorch should be carried on logits instead of softmax outputs

### ISSUE THREAD

**Issue by houxianxu** [IBM/adversarial-robustness-toolbox#75]
**Describe the bug**
The loss calculated in [loss_gradient](https://github.com/IBM/adversarial-robustness-toolbox/blob/267a6e6ec4d490e52c81986e91f933cea5670e15/art/classifiers/pytorch.py#L331) is based on the model_outputs[-1] and the corresponding label. However, the [model_outputs[-1]](https://github.com/IBM/adversarial-robustness-toolbox/blob/267a6e6ec4d490e52c81986e91f933cea5670e15/art/classifiers/pytorch.py#L560)  in Pytorch Wrapper is the softmax outputs. Should model_ouputs[0] be used if we use cross_entropy loss like in the [test of Pytorch ](https://github.com/IBM/adversarial-robustness-toolbox/blob/c65512b6ce64976a68fb2d9a19e132b5f6dddd0a/tests/classifiers/test_pytorch.py#L62)?

For convenient, related codes are shown below
1. The code to compute model_outputs:
```Python
result = []
if isinstance(self._model, nn.Sequential):
    for _, module_ in self._model._modules.items():
        x = module_(x)
        result.append(x)
elif isinstance(self._model, nn.Module):
    x = self._model(x)
    result.append(x)
else:
    raise TypeError("The input model must inherit from `nn.Module`.")
output_layer = nn.functional.softmax(x, dim=1)
result.append(output_layer)
return result
```

2. The code to compute the loss:
```python
# Compute the gradient and return
model_outputs = self._model(inputs_t)
loss = self._loss(model_outputs[-1], labels_t)
```

**Comment by minhitbk** [comment]
Hi @houxianxu :  Thanks alot for the issue you have opened. We have identified it and will fix it in the next few days.

**Comment by ririnicolae** [comment]
This issue will be solved later this week by the solution for epic #50.

**Comment by ririnicolae** [comment]
Fixed in #119 (on `dev` branch). Loss function can now be specified by the user for `PyTorch` classifier and will be applied directly on the output of the model.
